core/src/main/java/org/apache/iceberg/FileCleanupStrategy.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg;
+
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.function.Consumer;
+import org.apache.iceberg.avro.Avro;
+import org.apache.iceberg.exceptions.NotFoundException;
+import org.apache.iceberg.io.CloseableIterable;
+import org.apache.iceberg.relocated.com.google.common.base.Joiner;
+import org.apache.iceberg.util.Tasks;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@SuppressWarnings("checkstyle:VisibilityModifier")
+abstract class FileCleanupStrategy {
+
+  private final Logger log = LoggerFactory.getLogger(getClass());
+
+  protected final TableOperations ops;
+  protected final ExecutorService deleteExecutorService;
+  protected final ExecutorService planExecutorService;
+  protected final Consumer<String> deleteFunc;
+
+  protected FileCleanupStrategy(
+      TableOperations ops,
+      ExecutorService deleteExecutorService,
+      ExecutorService planExecutorService,
+      Consumer<String> deleteFunc) {
+    this.ops = ops;
+    this.deleteExecutorService = deleteExecutorService;
+    this.planExecutorService = planExecutorService;
+    this.deleteFunc = deleteFunc;
+  }
+
+  public abstract void cleanFiles();
+
+  protected static final Schema MANIFEST_PROJECTION =
+      ManifestFile.schema()
+          .select(
+              "manifest_path", "manifest_length", "added_snapshot_id", "deleted_data_files_count");
+
+  protected CloseableIterable<ManifestFile> readManifestFiles(String manifestListLocation) {
+    return Avro.read(ops.io().newInputFile(manifestListLocation))
+        .rename("manifest_file", GenericManifestFile.class.getName())
+        .classLoader(GenericManifestFile.class.getClassLoader())
+        .project(MANIFEST_PROJECTION)
+        .reuseContainers(true)
+        .build();
+  }
+
+  protected void deleteMetadataFiles(
+      Set<String> manifestsToDelete, Set<String> manifestListsToDelete) {
+    log.warn("Manifests to delete: {}", Joiner.on(", ").join(manifestsToDelete));
```

Comment:
https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/RemoveSnapshots.java#L538

This is still warn as it is currently, but I'm doubting if it really needs to be warn level? We are intentionally deleting these files, warn level implies that something is in a "not fatally wrong" which is not the case. It may mislead users. @rdblue @singhpk234 @jackye1995